src/logic/safe/transactions/gas.ts

```diff
@@ -116,12 +118,57 @@ export const getGasEstimationTxResponse = async (txConfig: {
 
     return new BigNumber(estimationData.substring(138), 16).toNumber()
   }
+  throw new Error('Error while estimating the gas required for tx')
+}
+
+const estimateGasWithInfura = async (txConfig: {
```

Comment:
`estimateGasWithRPCCall` or something around that? you can make the JSON_RPC call to any node that supports it.